.github/workflows/tests.yml

```diff
@@ -445,6 +445,153 @@ jobs:
           name: lxd-clients-${{ runner.os }}
           path: bin/
 
+  ui-e2e-tests:
+    name: UI e2e tests
+    runs-on: ubuntu-latest
+    needs: [code-tests, documentation]
+    if: ${{ ( github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || github.event_name == 'push' ) && github.ref_name == 'main' && github.repository == 'edlerd/lxd' }}
+    env:
+      LD_LIBRARY_PATH: "/home/runner/go/bin/dqlite/libs/"
+      LXD_DIR: "/var/lib/lxd"
+      LXD_OIDC_CLIENT_ID: ${{ secrets.LXD_UI_OIDC_CLIENT_ID }}
+      LXD_OIDC_ISSUER: ${{ secrets.LXD_UI_OIDC_ISSUER }}
+      LXD_OIDC_AUDIENCE: ${{ secrets.LXD_UI_OIDC_AUDIENCE }}
+      LXD_OIDC_USER: ${{ secrets.LXD_UI_OIDC_TEST_USER }}
+      LXD_OIDC_PASSWORD: ${{ secrets.LXD_UI_OIDC_TEST_PASSWORD }}
+      LXD_OIDC_GROUPS_CLAIM: "lxd-idp-groups"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Checkout LXD-UI
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: 'canonical/lxd-ui'
+          ref: main
+          path: lxd-ui
+
+      - name: Install Dotrun
+        run: sudo pip3 install dotrun
+
+      - name: Install LXD-UI dependencies
+        run: |
+          set -eux
+          sudo chmod 0777 ./lxd-ui
```

Comment:
out of interest how come this is needed?

Reply:
To be honest, this is pretty old code and I don't remember the exact details. I assume `dotrun` is creating files and folders under a different user, and we need to give it permission to do so.